projects/ion/src/lib/pagination/pagination.component.ts

```diff
@@ -136,6 +153,52 @@ export class IonPaginationComponent implements OnChanges, OnInit {
     return (option && option.label) || this.generateLabel(this.itemsPerPage);
   }
 
+  visibleButtons(): Page[] {
+    const pagesLength = this.pages.length;
+    const currentPageIndex = this.pages.indexOf(this.currentPage());
+    const visiblePageCount = Math.min(5, pagesLength);
+
+    let startPageIndex = Math.max(
+      1,
+      currentPageIndex - Math.floor(visiblePageCount / 2)
+    );
+    let endPageIndex = Math.min(
+      startPageIndex + visiblePageCount - 1,
+      pagesLength - 2
+    );
+
+    if (currentPageIndex < 3) {
+      endPageIndex = 4;
+    }
+    if (currentPageIndex > pagesLength - 3) {
+      startPageIndex = pagesLength - 5;
+    }
+
+    return this.pages.slice(startPageIndex, endPageIndex + 1);
+  }
+
+  jumpPagesFoward(): void {
+    this.hoverControl.right = false;
+    if (this.currentPage().page_number >= this.pages.length - 4) {
+      this.selectPageOnClick(this.pages.length);
+      return;
+    }
+    this.selectPageOnClick(this.currentPage().page_number + 5);
```

Comment:
Since 5 is a significant value to advanced pagination, you can name it too. Something like "DEFAULT_VISIBLE_PAGES_WHEN_ADVANCED" (not a good name, but you got the idea).